src/Azure/Orleans.ScheduledJobs.AzureStorage/AzureStorageJobShard.cs

```diff
@@ -23,34 +28,40 @@ public AzureStorageJobShard(string id, DateTimeOffset startTime, DateTimeOffset
         BlobClient = blobClient;
         ETag = eTag;
         Metadata = metadata;
+        
+        // Create unbounded channel for storage operations
+        // In the future, we could add batching here
+        _storageOperationChannel = Channel.CreateUnbounded<StorageOperation>(new UnboundedChannelOptions
+        {
+            SingleReader = true,
+            SingleWriter = false
+        });
+        
+        // Start the background task that processes storage operations
+        _storageProcessorTask = ProcessStorageOperationsAsync();
```

Comment:
Do we know which context this is constructed in? I'd be concerned that it could be capturing some async context or running on the wrong scheduler.